### ISSUE TITLE
Bad result when inference on pretraind model with custom dataset for spatio-temporal action detection

### ISSUE THREAD

**Issue by HAMA-DL-dev** [open-mmlab/mmaction2#2192]
## Check List

1. I have searched related issues but cannot get the expected help.
2. I have read [related issues, such as 'Worse results after train on custom classes?'](https://github.com/open-mmlab/mmaction2/issues/1802), but cannot get the expected help.

## my custom dataset and configuration 
I uploaded these a day before [on issue](https://github.com/open-mmlab/mmaction2/issues/2184#issuecomment-1386582727)

## Result 
Yesterday, I succeed to inference walking pedestrian video on SlowFast abd SlowOnly config based on AVA dataset. 
But fail to inference tail light video on pre-trained model based pm my custom dataset.

To summarize, the location of predicted bounding box was weird and another inference output video has no bounding box which means there is no object detected. CSV file seems to be normal, but I want to get your advice out the result below.

### video
![image](https://user-images.githubusercontent.com/59410853/213632466-485581c4-f228-45ca-95f5-4d3a492b852a.png)

### part of csv
![image](https://user-images.githubusercontent.com/59410853/213633877-5b6f2708-7796-4dad-ad0f-08a9755508dd.png)

## Trial

Using command $python demo/demo_spatiotemporal_det.py --video {inference_video.mp4} --configs configs/{my_config} --checkpoint {my_checkpoint.pth} --det-score-thr 0.8     --action-score-thr 0.5     --label-map ${my_data_annotaions}/label_map.txt     --predict-stepsize 8 --output-stepsize 4     --output-fps 6

## Questions and my suspects

1. I did not used **--validate** option when I train model cause of error : AttribueError at ${mmaction_envs}/lib/python3.7/site-packages/torch/utils/data/dataset.py", line 83, in __getattr__. Is this can be result on bad training output?

2. Is bounding box value format [x1, y1, x2, y2] or [x_center, y_center, w, h] ? My proposal file contains bounding box information not only coordinates of bbox but also confidence value and I get those using YOLOv5 with '--save-txt' and '--save-conf' options. The model returns text file of bounding box and its format follows latter, [x_center, y_center, w, h].

**Comment by mareksubocz** [comment]
Hi, as far as I've experienced, mmaction uses [x1, y1, x2, y2] format in most of its implementations.

**Comment by HAMA-DL-dev** [comment]
Sorry for not reading description [here](https://github.com/open-mmlab/mmaction2/blob/master/tools/data/ava/AVA_annotation_explained.md). I read this just after uploaded this issue and modified annotation data. But nothing had been changed after training modified annotations. 

The task that I make use of tail light detection is spatio-temporal action detection. So I inference a custom trained model refer to [this documentation](https://mmaction2.readthedocs.io/en/latest/demo.html#spatiotemporal-action-detection-video-demo). 
![image](https://user-images.githubusercontent.com/59410853/219933848-297eba8a-a062-4720-a4af-57167c486218.png)

There is need to configure '--det-config' and '--det-checkpoint' and I configure this options default. But my model focus on detecting car, especially its tail light. So I suspect the bad result problem cause of incorrect configurations, '--det-config' and '--det-checkpoint'. Do I need another custom training for two configurations via mmdetection?

**Comment by eliethesaiyan** [comment]
@HAMA-DL-dev , i think in order to do spatial temporal inference on your custom dataset(car-tail-light), you will need two models(2config files), one for detection from mmdetect and another from mmaction2.
```--det-config``` should points to your mmdetect config file and ```--det-checkpoint``` to the mmdetect checkpoint file.
From what i can deduce from ```demo_spatial_temporal_det.py``` code, action detection should happen only when detection happened.

**Comment by HAMA-DL-dev** [comment]
@eliethesaiyan 
Thanks to your advise. I tried to train ```mmdetection``` based on by custom dataset. The result looks nicer than before. But there is still some problems. After I succeed custom train perfectly, I will let you know the custom train process and the purpose of using proposal value (cause I check your comment asking a usage of proposal file)

**Comment by Harry-KIT** [comment]
Hi @HAMA-DL-dev 
Can you give me some feedback about how to annotate and prepare custom dataset like AVA?
Thank you in advance
